### ISSUE TITLE
Random issue with register creating cash refund

### ISSUE THREAD

**Issue by printgeek** [opensourcepos/opensourcepos#3081]
### Background information

__IMPORTANT: If you choose to ignore this issue report template, your issue will be closed as we cannot help without the requested information.__

Please make sure you tick (add an x between the square brackets with no spaces) the following check boxes:

- [X] Reporting an issue of an unmodified OSPOS installation
- [X] Checked [open and closed issues](https://github.com/opensourcepos/opensourcepos/issues?utf8=%E2%9C%93&q=is%3Aissue) and no similar issue was already reported (please make sure you searched!)
- [X] Read [README](https://github.com/opensourcepos/opensourcepos/blob/master/README.md), [WHATS_NEW](https://github.com/opensourcepos/opensourcepos/blob/master/WHATS_NEW.txt), [INSTALL.md](https://github.com/opensourcepos/opensourcepos/blob/master/INSTALL.md) and [UPGRADE](https://github.com/opensourcepos/opensourcepos/blob/master/UPGRADE.txt)
- [X] Read the [FAQ](https://github.com/opensourcepos/opensourcepos#faq) for any known install and/or upgrade gotchas (in specific PHP extensions installed)
- [X] Read the [wiki](https://github.com/opensourcepos/opensourcepos/wiki)
- [X] Executed any database upgrade scripts if an upgrade pre 3.0.0 (e.g. database/2.4_to_3.0.sql)
- [X] Aware the installation code is in [bintray](https://bintray.com/jekkos/opensourcepos/opensourcepos/view/files?sort=updated&order=asc#files) (see README), and [GitHub master](https://github.com/opensourcepos/opensourcepos/tree/master) is for [developers only](https://github.com/opensourcepos/opensourcepos/wiki/Development-setup) and therefore not complete nor stable

### Installation information

- OSPOS version is: 3.3.3
- OSPOS git commit hash is: 
- PHP version is: 7.3.19-1~deb10u1
- MySQL or MariaDB version is: 5.5.5-10.3.27-MariaDB-0+deb10u1
- OS and version is: Linux 5.4.83-v7+
- WebServer is: Apache/2.4.38 (Raspbian)
- Selected language is: en-US
- (If applicable) Docker installation: N/A
- (If applicable) Installation package for the LAMP/LEMP stack is: N/A

### Issue / Bug  / Question / New Feature

We've had a random recurring issue on the sales screen and I've been unable to track down the root cause. It seems to have started after I updated to 3.3.0 last year. What happens is that every so often (once in every ~50 transactions) we will press the 'Add Payment' button, regardless of the payment type, the screen will blank momentarily and when it returns all of the items are gone from the register. Then, when you proceed to reenter the items and complete the sale what will show in the payment amount is something like "Credit Card 62.48, Cash -31.24", where the sale amount is $31.24, but it shows instead that the transaction was for 2x the amount, and a cash refund of the amount was given. This is how it looks in the database:
![Screen Shot 2021-01-18 at 11 48 42 AM](https://user-images.githubusercontent.com/4838571/104948578-23bb5700-5983-11eb-9274-066cd93ffa55.png)

Any ideas what could be causing this? As you can see I am running this on a raspberry pi and the database is starting to get pretty large, so I am in the process of purchasing something with a bit more giddyup in case its a performance issue. Would it be a problem with Mysql or PHP configuration?

Also, keep in mind I've never seen this happen, I'm having to get details from the people that run the register when it does, all I see is what is in the reports/database.

tl;dr: register is randomly cutting out and the result is a transaction that looks like someone charged double and the pocketed the change.

**Comment by odiea** [comment]
Have you tried cancelling the Sale before you reenter the items?

**Comment by jekkos** [comment]
@printgeek does this occur when you add a lot of items to the register? Something was fixed recently that you might want to tryout. #3068

**Comment by jekkos** [comment]
Otherwise you need to assist in reproducing this as the description is too vague.

**Comment by printgeek** [comment]
> 
> 
> Have you tried cancelling the Sale before you reenter the items?

Yes, that works. If the employee scans that first item, sees the payment amount pop up mysteriously, and realizes what happened, they can cancel the sale and restart from scratch with no issues. We've tried to train them to recognize it, and they've gotten pretty good at it, but I would definitely like to find the cause. It's also not too difficult to query for and fix it in the database after the fact, it's all in that ospos_sales_payments table, so after I find the cause I'll go in and correct for it.

**Comment by printgeek** [comment]
> 
> 
> does this occur when you add a lot of items to the register? Something was fixed recently that you might want to tryout. #3068 . This change is in latest master (download latest unstable from bintray).

@jekkos No, in fact the example above was a single item, but I'll look at that patch for some clues, thanks.

> 
> 
> Otherwise you need to assist in reproducing this as the description is too vague.

I certainly will, I have been trying to trace what changes I may have made to the configuration in November and troubleshooting there first. Tracking down an intermittent problem is never fun, so I thought I'd reach out to the community and see if anyone had seen this or had recommendations for tools I could use to monitor it. I have a sandbox set up with a copy of the database, I figure I'll keep creating sales until it happens, but will the developer tools in Chrome give me anything useful, or is there a tool I can run on the server to monitor activity or even Apache/PHP/MySql logs that I can sift through?

Any input is greatly appreciated.

**Comment by SteveIreland** [comment]
@printgeek is this still an issue for you?

**Comment by printgeek** [comment]
@SteveIreland After replacing the server and the computer we were using as the register it's no longer an issue. We *think* the problem was caused by a combination of slow server and a "bouncy" touch screen, so that when the add payment button was pressed on the touchscreen, the screen bounced and caused the button to be pushed a second time before the screen refreshed. Could have also had to do with the employees with longer nails, funnily enough.

I think we can put this one to bed unless you think it's worth looking into that screen refresh.

**Comment by SteveIreland** [comment]
Since I've done some work in this area I'm hoping that if there was a problem then it is resolved.  So I'll go ahead and close this one.
